src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/ModelMetadataIdentity.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public struct ModelMetadataIdentity
+    {
+        public static ModelMetadataIdentity ForType([NotNull] Type modelType)
+        {
+            return new ModelMetadataIdentity()
+            {
+                ModelType = modelType,
+                MetadataKind = ModelMetadataKind.Type,
+            };
+        }
+
+        public static ModelMetadataIdentity ForParameter([NotNull] ParameterInfo parameterInfo)
+        {
+            return new ModelMetadataIdentity()
+            {
+                ParameterInfo = parameterInfo,
+                ModelType = parameterInfo.ParameterType,
+                Name = parameterInfo.Name,
+                MetadataKind = ModelMetadataKind.Parameter,
+            };
+        }
+
+        public static ModelMetadataIdentity ForProperty(
+            [NotNull] Type modelType,
```

Comment:
Unfortunately this can't work in CoreCLR. A CoreCLR `PropertyInfo` doesn't know it's declaring type.